src/udev/udev-builtin-net_id.c

```diff
@@ -794,6 +797,44 @@ static int names_mac(sd_device *dev, struct netnames *names) {
         return 0;
 }
 
+static int names_netdevsim(sd_device *dev, struct netnames *names) {
+        sd_device *netdevsimdev;
+        const char *sysname;
+        unsigned int addr;
+        const char *port_name = NULL;
+        size_t l;
+        char *s;
+        int r;
+
+        assert(dev);
+        assert(names);
+
+        r = sd_device_get_parent_with_subsystem_devtype(dev, "netdevsim", NULL, &netdevsimdev);
+        if (r < 0)
+                return r;
+
+        r = sd_device_get_sysname(netdevsimdev, &sysname);
+        if (r < 0)
+                return r;
+
+        if (sscanf(sysname, "netdevsim%u", &addr) != 1)
+                return -EINVAL;
```

Comment:
I'd prefer
```C
const char *t;
t = startswith(t, "netdevsim");
if (!t)
     return -EINVAL;
r = safe_atou(t, &addr);
if (r < 0)
     return log_debug_errno(r, "Failed to parse netdevsim device number %s: %m", t);
```
This does better error handling, in particular trailing garbage is detected properly.

Reply:
Okay, again, I just copy&pasted and adjusted the existing code. Will do.

Reply:
Hmm, your code tries to read uint from the beginning of the string :/

Reply:
I don't know, is this really needed? The rest of the code just uses sscanf. The input is sysfs file name, not user input. Looks like overkill to me...

Reply:
I fixed the error in the snippet. I guess sscanf is ok in this particular case.